flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/tasks/StreamTaskMailboxTestHarnessBuilder.java

```diff
@@ -163,6 +163,7 @@ public StreamTaskMailboxTestHarnessBuilder(
         streamMockEnvironment.setTaskMetricGroup(taskMetricGroup);
 
         StreamTask<OUT, ?> task = taskFactory.apply(streamMockEnvironment);
+        task.restore();
         task.beforeInvoke();
```

Comment:
I'm a bit confused. Do we have both `restore()` and `beforeInvoke()` now? Why do we need both?

Reply:
beforeInvoke is still needed for test purposes. It helps to switch the internal isRunning flag to true without actual call the invoke. But if you think that it is problem, I can try to rewrite the test.